cartographer_ros/cartographer_ros/offline_node.cc

```diff
@@ -21,37 +21,41 @@
 #include <sys/resource.h>
 #include <time.h>
 #include <chrono>
-#include <queue>
 
 #include "cartographer_ros/node.h"
+#include "cartographer_ros/playable_bag.h"
 #include "cartographer_ros/split_string.h"
 #include "cartographer_ros/urdf_reader.h"
 #include "ros/callback_queue.h"
-#include "rosbag/bag.h"
-#include "rosbag/view.h"
 #include "rosgraph_msgs/Clock.h"
-#include "tf2_msgs/TFMessage.h"
 #include "tf2_ros/static_transform_broadcaster.h"
 #include "urdf/model.h"
 
 DEFINE_string(configuration_directory, "",
               "First directory in which configuration files are searched, "
               "second is always the Cartographer installation to allow "
               "including files from there.");
-DEFINE_string(configuration_basename, "",
-              "Basename, i.e. not containing any directory prefix, of the "
-              "configuration file.");
-DEFINE_string(bag_filenames, "", "Comma-separated list of bags to process.");
+DEFINE_string(configuration_basenames, "",
+              "Basenames, i.e. not containing any directory prefix, of the "
+              "configuration files for each trajectory.");
 DEFINE_string(
-    urdf_filename, "",
-    "URDF file that contains static links for your sensor configuration.");
+    bag_filenames, "",
+    "Comma-separated list of bags to process. One bag per trajectory.");
+DEFINE_string(urdf_filenames, "",
```

Comment:
Is this optional? If yes, please note in the description.

Reply:
It depends on the bag. If your range data is in the tracking frame, or the transforms are in the bag, then it is optional. If this is not the case, you need to provide an urdf.